### PR TITLE
feat: [PIE-5729]: Trigger Drawer Catalog API Integration  

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger jest

**Review by kapilkumawat86 — DISMISSED** [review]

**Review by samarth-sharma31 — DISMISSED** [review]

File: src/modules/10-common/featureFlags.ts [hunk 1]
```diff
@@ -125,5 +125,6 @@ export enum FeatureFlag {
   ASG_NG = 'ASG_NG',
   CDS_ARTIFACTORY_REPOSITORY_URL_MANDATORY = 'CDS_ARTIFACTORY_REPOSITORY_URL_MANDATORY',
   NG_K8_COMMAND_FLAGS = 'NG_K8_COMMAND_FLAGS',
-  CDC_DASHBOARD_ENHANCEMENT_NG = 'CDC_DASHBOARD_ENHANCEMENT_NG'
+  CDC_DASHBOARD_ENHANCEMENT_NG = 'CDC_DASHBOARD_ENHANCEMENT_NG',
+  NG_TRIGGER_LIST_API_ENABLED = 'NG_TRIGGER_LIST_API_ENABLED'
```
Comment (pankajsingh-harness) [inline src/modules/10-common/featureFlags.ts:129]:
```suggestion
  CD_TRIGGER_CATALOG_API_ENABLED = 'CD_TRIGGER_CATALOG_API_ENABLED'
```

**Comment by pankajsingh-harness** [comment]
retrigger cypress-pipeline

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger jest

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger coverage

**Comment by pankajsingh-harness** [comment]
retrigger coverage

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cv

**Comment by pankajsingh-harness** [comment]
trigger featurenamecheck

**Review by kapilkumawat86 — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**